File: server/controllers/MedalPage.js
const MedalPageDB = require('./../models/MedalPageDB/MedalPageDB.js');


async function MedalTally(req,res){
    try {
        let medalRecords = await MedalPageDB.query(`SELECT COUNTRY, GOLD, SILVER, BRONZE FROM MEDALPAGE ORDER BY GOLD, SILVER, BRONZE DESC;`);
        medalRecords = medalRecords.rows;

        if(medalRecords.length > 0){
            res.send(medalRecords.rows);
        }else {
            res.send(`None of the countries won any medal`);
        }
    } catch (err) {
        res.send("Failed to fetch medal tally");
    }
}

async function UpdateMedalTally(req, res){
   
    try {
        const {COUNTRY, MEDALTYPE, ADDNEWMEDAL} = req.body;
    
        let UpdatedRecords = await SchedulePageDB.query(` UPDATE MedalPage
        SET  ${MEDALTYPE} +=  ADDNEWMEDAL
        WHERE COUNTRY = ${COUNTRY} AND MEDALTYPE = ${MEDALTYPE}; `);
        
        UpdatedRecords = UpdatedRecords.rows;
        if(UpdatedRecords.length > 0){
            res.send(UpdatedRecords);
        }else {
            res.send(`No records were updated`);
        }
    
    } catch (err) {
        res.send("Failed to fetch schedules");
    }
}


module.exports = {MedalTally, UpdateMedalTally};

File: Olympics-fanone-in-golang-gin/controllers/MedalPage.go
package controllers;
import (
	// "log"
	// "strconv"
	// "time"
	"gitlab.com/satvikshrivas26/olympic-fanzone/models"
	"github.com/gin-gonic/gin"
	_ "github.com/go-sql-driver/mysql"
)


func MedalTally(c *gin.Context) {
	
	var medalRecords []models.Medal
	_, err := dbmap.Select(&medalRecords, "SELECT * FROM MEDAL ORDER BY GOLD, SILVER, BRONZE DESC;");
		if err != nil {
			c.JSON(500, gin.H{"error": "Failed to fetch medal tally"})
		}else if(len(medalRecords) > 0){
			c.JSON(200, medalRecords)
		}else {
			c.JSON(200, gin.H{"message": "None of the countries won any medal"})
		}
}
func UpdateMedalTally(c *gin.Context){

	// _, err := dbmap.Select(&UpdatedRecords, " SELECT * FROM SCHEDULE WHERE CAST(STARTDATE AS DATE) <= CURDATE() AND CAST(ENDDATE AS DATE) >= CURDATE();");
	var UpdatedRecords []models.Medal
	
		if err := c.ShouldBindJSON(&UpdatedRecords); err != nil {
			c.JSON(404, gin.H{"error": err.Error()})
			return
		}

}